src/App.tsx

```diff
@@ -1,24 +1,125 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { addTodos, deleteTodo, getTodos } from './api/todos';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Notification } from './components/Notifications';
+import { TodoList } from './components/TodoList';
+import { ErrorMessages } from './types/errorMessages';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+import { getVisibleTodos } from './utils/PreparedTodo';
 
-const USER_ID = 0;
+const USER_ID = 6386;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasServerError, setServerError] = useState(false);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [name, setName] = useState('');
+  const [errorType, setErrorType] = useState<ErrorMessages>(ErrorMessages.NONE);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const visibletodos = getVisibleTodos(todos, status);
+
+  const getTodosFromServer = async () => {
+    try {
+      setServerError(false);
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setServerError(true);
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const handleAddTodo = async (todoName: string) => {
+    if (todoName.length === 0) {
+      setErrorType(ErrorMessages.TITLE);
+      setServerError(true);
+
+      return;
+    }
+
+    const todoToAdd = {
+      id: 0,
+      userId: USER_ID,
+      title: todoName,
+      completed: false,
+    };
+
+    const newTodo = await addTodos(USER_ID, todoToAdd);
+
+    setTempTodo(newTodo);
+
+    try {
+      setName('');
+      setTempTodo(null);
+
+      await getTodosFromServer();
```

Comment:
Your `todoToAdd ` shouldn't have an id, it will be set on the server. You should show demo-todo with the loader while waiting for loading the actual todo from the server.